platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java

```diff
@@ -362,6 +363,37 @@ public String getFileContent(@PathParam("machineId") String machineId,
         return machineManager.getMachine(machineId).readFileContent(path, startFrom, limit);
     }
 
+    /**
+     * Copies a resource in the filesystem from specified container.
+     *
+     * @param sourceMachineId
+     *         source machine id
+     * @param sourcePath
+     *         path to source file or directory inside specified container
+     * @param targetPath
+     *         path to destination file or directory inside container
+     * @param overwrite
+     *         If "false" then it will be an error if unpacking the given content would cause
+     *         an existing directory to be replaced with a non-directory or other resource and vice versa.
+     * @throws MachineException
+     *         if any error occurs when resources are copying into the container
+     * @throws NotFoundException
+     *         if any error occurs with getting source machine
+     */
+    @POST
+    @Path("/{machineId}/filepath/{sourcePath:.*}")
+    @RolesAllowed("user")
+    public void copyFilesBetweenMachines(@PathParam("machineId") String sourceMachineId,
+                                         @PathParam("sourcePath") String sourcePath,
+                                         @QueryParam("targetMachineId") String targetMachineId,
+                                         @QueryParam("targetPath") String targetPath,
+                                         @DefaultValue("false") @QueryParam("overwrite") Boolean overwrite)
+            throws NotFoundException, ServerException, ForbiddenException {
+        checkCurrentUserPermissions(machineManager.getMachine(sourceMachineId));
+
+        machineManager.getMachine(targetMachineId).copy(machineManager.getMachine(sourceMachineId), sourcePath, targetPath, overwrite);
```

Comment:
it is not needed because when machine id null we throw exception machine with that id not found and if path null we have exception from docker api  'path' cannot be empty